### PR TITLE
General inclusions

### PR DESCRIPTION
**Pull request by apozharski**
General inclusions without lifting

### CODE REVIEW

File: nosnoc/nosnoc.py [hunk 1]
```diff
@@ -44,12 +46,14 @@ class NosnocModel:
...
                  x: ca.SX,
-                 F: List[ca.SX],
                  x0: Optional[np.ndarray],
+                 F: Optional[List[ca.SX]] = None,
                  c: Optional[List[ca.SX]] = None,
                  S: Optional[List[np.ndarray]] = None,
                  g_Stewart: Optional[List[ca.SX]] = None,
                  u: ca.SX = ca.SX.sym('u_dummy', 0, 1),
+                 alpha: ca.SX = ca.SX.sym('alpha_dummy', 0, 1),
+                 f_x: Optional[List[ca.SX]] = None,
```
[the remark this answers is not in the document]
Comment (FreyJo) [reply nosnoc/nosnoc.py:56]: As far as I understand it should be something like:

`\dot{x} \in \sum \alpha_i f_x(x, alpha, u,p_time_var, p_global, v_global)`
where
`\alpha_i \in [0, 1]
\alpha_i = 0 if c_i < 0
\alpha_i = 1 if c_i > 0`
and FESD complementarities are enforced on `alpha`, `1-alpha` such that they can only change between  finite elements.

**Review by FreyJo — COMMENTED** [review]
Great work @apozharski !
I added some minor comments.
This new way of setting up a model is surely more involved, but more flexible.
I guess you discussed some of this with @nurkanovic 
I wonder how the name "general inclusions" is motivated.

File: nosnoc/nosnoc_opts.py [hunk 2]
```diff
@@ -18,6 +18,8 @@ class NosnocOpts:
     print_level: int = 0  #: higher -> more info
     max_iter_homotopy: int = 0
 
+    general_inclusion: bool = False
```
[the remark this answers is not in the document]
Comment (FreyJo) [reply nosnoc/nosnoc_opts.py]: outdated.

**Comment by nurkanovic** [comment]
> 

The reason for the name general inclusions is that when you play with expressions of `alpha` in the r.h.s. of the ODE/DI
you get sets that are not Filippov or not even convex sets, hence it is a more general inclusion. 
Figure 2 in Acary2014 illustrates this nicely.

**Review by FreyJo — APPROVED** [review]
Looks all good now!

**Pull request merged**